l2l/optimizers/multievolution/optimizer.py

```diff
@@ -0,0 +1,239 @@
+        # Structure initializers
+        toolbox.register("individual", tools.initIterate, creator.Individual,
+                         lambda: dict_to_list(optimizee_create_individual()))
+        toolbox.register("population", tools.initRepeat, list, toolbox.individual)
+
+        # Operator registering
+        # This complex piece of code is only necessary because we're using the
+        # DEAP framework and would like to decorate the DEAP mutation operator
+        def bounding_decorator(func):
+            def bounding_wrapper(*args, **kwargs):
+                if self.optimizee_bounding_func is None:
+                    return func(*args, **kwargs)
+                else:
+                    # Deap Functions modify individuals in-place, Hence we must do the same
+                    result_individuals_deap = func(*args, **kwargs)
+                    result_individuals = [list_to_dict(x, self.optimizee_individual_dict_spec)
+                                          for x in result_individuals_deap]
+                    bounded_individuals = [self.optimizee_bounding_func(x) for x in result_individuals]
+                    for i, deap_indiv in enumerate(result_individuals_deap):
+                        deap_indiv[:] = dict_to_list(bounded_individuals[i])
+                    print("Bounded Individual: {}".format(bounded_individuals))
+                    return result_individuals_deap
+
+            return bounding_wrapper
+
+        toolbox.register("mate", tools.cxBlend, alpha=parameters.mate_par)
+        toolbox.decorate("mate", bounding_decorator)
+        toolbox.register("mutate", tools.mutGaussian, mu=0, sigma=parameters.mut_par, indpb=traj.ind_prob)
+        toolbox.decorate("mutate", bounding_decorator)
+        toolbox.register("select", tools.selTournament, tournsize=traj.tourn_size)
+
+        # ------- Initialize Population and Trajectory -------- #
+        # NOTE: The Individual object implements the list interface.
+        self.pop = toolbox.population(n=traj.pop_size)
+        self.eval_pop_inds = [ind for ind in self.pop if not ind.fitness.valid]
+        self.eval_pop = [list_to_dict(ind, self.optimizee_individual_dict_spec)
+                         for ind in self.eval_pop_inds]
+        self.eval_pop = self.compress_individual(self.eval_pop, traj.n_inner_params)
+        self.g = 0  # the current generation
+        self.toolbox = toolbox  # the DEAP toolbox
+        self.hall_of_fame = HallOfFame(20)
+        self.best_individual = None
+
+        self._expand_trajectory(traj)
+
+
+    def expand_individual(self, c_population, inner_params):
+        individual_exp = [{} for i in range(len(c_population)*inner_params)]
+        for ind_id, elem in enumerate(c_population):
+            for key in self.grouped_params_dict.keys():
+                parameters = elem[key]
+                for ix, e in enumerate(parameters):
+                    individual_exp[ind_id*inner_params+ix][key] = float(e)
```

Comment:
Are we sure the parameters are always castable to float? E.g. could parameters be a list?

Reply:
This is a good point... I am not sure. We can try to cast it to a float and otherwise leave it as is... not sure what will happen by leaving it like that though.